sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -1544,11 +1544,7 @@ case class GpuCast(
     }
   }
 
-  override def toString: String = if (ansiMode) {
```

Comment:
Can we add a test for this? I think this is something that needs to go into the shim layer since it varies by Spark version. 

This is the Spark 3.1.x implementation:

```
  override def toString: String = {
    val ansi = if (ansiEnabled) "ansi_" else ""
    s"${ansi}cast($child as ${dataType.simpleString})"
  }
```